File: backend/src/main/kotlin/com/casadosreclamos/dto/UserDto.kt
package com.casadosreclamos.dto

import com.casadosreclamos.model.Role
import com.casadosreclamos.model.User

data class UserDto(val email: String?, val name: String?, val roles: MutableSet<Role>, val credits: MutableSet<PlafondDto>?) {
    constructor(user: User) : this(
        user.email,
        user.name,
        user.roles,
        user.credits.stream().map { PlafondDto(it) }.toList().toMutableSet()
    )
}

File: frontend/src/models/UserDto.ts
class PlafondDetail {
  brand: string;
  amount: number;

  constructor(obj: PlafondDetail) {
    this.brand = obj.brand;
    this.amount = obj.amount;
  }
}

export default class UserDto {
  email: string | null = null;
  name: string | null = null;
  credits: Array<PlafondDetail> | null = null;
  roles: Array<string> | null = null;

  constructor(obj?: UserDto) {
    if (obj) {
      this.email = obj.email!!;
      this.name = obj.name!!;
      this.credits = obj.credits!!.map((element) => new PlafondDetail(element));
      this.roles = obj.roles!!;
    }
  }
}
